SHARDING.md

```diff
@@ -0,0 +1,464 @@
+
+Update the TrustedRoot for the new shard's key material:
+
+```
+{
+  "mediaType": "...",
+  "tlogs": [
+    // new shard
+    {
+      "baseUrl": "https://<year-revions>.rekor.(sigstore|sigstage).dev",
+      "hashAlgorithm": "SHA2_256",
+      "publicKey": {
+        "rawBytes": "base64-encoded DER public key",
+        "keyDetails": "ED25199",
+        "validFor": {
+          "start": "<UTC timestamp for when shard was spun up>"
+        }
+      },
+      "logId": {
+        "keyId": "wNI9atQGlz+VWfO6LRygH4QUfY/8W4RFwiT5i5WRgB0="
+      }
+    },
+    // previous shard
+    {
+      "baseUrl": "...",
+      "hashAlgorithm": "SHA2_256",
+      "publicKey": {
+        "rawBytes": "...",
+        "keyDetails": "...",
+        "validFor": {
+          "start": "...",
+          "end": "<UTC timestamp that is after the shard is turned read-only>"
+        }
+      },
+      "logId": {
+        "keyId": "..."
+      }
+    },
+    ...
+  ],
+  ...
+}
+```
+
+
+Update the SigningConfig for the new shard:
+
+```
+{
+    "media_type": "...",
+    "ca_urls": [...],
+    "oidc_urls": [...],
+    // sorted from newest to oldest
+    "rekor_tlog_urls": [
+        // new shard
+        {
+            "url": "<year-revision>.rekor.(sigstore|sigstage).dev",
+            "major_api_version": 2,
+             "valid_for": {
+                "start": "<UTC timestamp, at least one week past when to-be-signed TUF timestamp will expire>",
+            }
+        },
+        // previous shard
+        {
+            ...,
+            "valid_for": {
+                "start": "...",
+                "end": "<valid_for.start for new shard>"
+            }
+        }
+    ],
+    "rekor_tlog_config": {...},
+    "tsa_urls": [...],
+    "tsa_config": {...},
+}
+```
+
+Follow the playbooks in https://github.com/sigstore/root-signing and
+https://github.com/sigstore/root-signing-staging to orchestrate a new
+signing event.
+
+## Turn Down Old Shard
+
+Wait a week, until all clients have received updated SigningConfigs
+and started signing using the new shard, and are using the updated TrustedRoot
+to verify inclusion proofs.
+
+### Verify write traffic
+
+Verify write traffic to the previous shard has gone down to ~0 QPS.
+If there is significant traffic, look at the user agent and see if you
+can figure out who's calling the service.
+
+<!--- TODO: Update with link to monitoring -->
+
+### Post on Slack
+
+Make a post on Slack letting the community know we'll be freezing
+and turning down the log.
+
+### WIP: Disable alerts for old shard
+
+Make sure to either disable or remove via Terraform alerts for the old shard
+for failing healthchecks for the write path or missing metrics. As we turn down the service
+and delete resources, we expect that healthchecks will start to fail.
+
+As we create alerting, we'll finalize this section with what alerts need to be removed.
+
+### Freeze the log
+
+Pass `--read-only=true` to the service, which will stop write traffic and
```

Comment:
I think we could skip this step and just turn off the K8s deployment.

Reply:
I'm good with that. The only benefit of freezing the log is to gracefully reject write traffic, but unless we're adding in a waiting period for clients to fail and update to the latest shard (which shouldn't be an issue as clients should be using TUF), this doesn't add much.